annotations/templates/annotations/relationtemplate_list.html

```diff
@@ -26,16 +26,61 @@
                     </tr>
                 </thead>
                 <tbody>
                     {% for template in data.templates %}
                     <tr class='clickable-row' data-href='{% url 'get_relationtemplate' template.id %}'>
+                        <td>
+                            <a class="fa fa-trash"
+                               aria-hidden="true"
+                               data-toggle="modal"
+                               data-delete-template-tooltip="tooltip"
+                               title="Delete relation template"
+                               id="openTemplateDeleteModal{{template.id}}"
+                               data-id = "#TemplateDeleteModal{{template.id}}"
+                            />
+                        </td>
                         <td>{{ template.name }}</td>
                         <td>{{ template.description }}</td>
                     </tr>
+
+                    <!-- Delete Template modal section starts here -->
+                    <div class="modal fade" id="TemplateDeleteModal{{template.id}}" tabindex="-1" role="dialog" aria-labelledby="deleteModalLabel">
+                      <div class="modal-dialog" role="document">
+                        <div class="modal-content">
+                          <div class="modal-header">
+                            <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+                              <h4 class="modal-title" id="deleteModalLabel"><strong>Delete Relation Template</strong></h4>
+                          </div>
+
+                          <form action=" {% url "delete_relationtemplate" template.id %}" method="post">
+                              {% csrf_token %}
+                          <div class="modal-body">
+                            <h4>
+                              Are you sure you want to delete the following relation template: <br><br>
+                              <div class="form-group">
+                                <strong>Name:</strong> <br><label style="font-weight:normal" >{{template.name}}</label>
+                              </div>
+
+                              <div class="form-group">
+                                <strong>Description:</strong> <br> <label style="font-weight:normal">{{template.description}}</label>
+                              </div>
+                            </h4>
+                          </div>
+                          <div class="modal-footer">
+                            <input class="btn btn-danger" type="submit" value="Delete">
+                            <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+                          </div>
+                          </form>
+                        </div>
+                      </div>
+                    </div>
+                    <!-- Delete Template modal section ends here -->
```

Comment:
I wouldn't create a modal for every single template. That just becomes unmanageable when there are many templates. Rather have one modal for which you set the selected template id when the delete button is clicked.

Reply:
Changed the implementation of creating multiple modals to a single modal.